bin/as.sh

```diff
@@ -385,15 +385,15 @@ attach_jvm()
 
 sanity_check() {
     # 0 check whether the pid exist
-    local pid=$(ps -p ${TARGET_PID} -o pid=)
+    local pid=$(ps -o pid= | grep " ${TARGET_PID}$")
     if [ -z ${pid} ]; then
         exit_on_err 1 "The target pid (${TARGET_PID}) does not exist!"
     fi
 
     # 1 check the current user matches the process owner
     local current_user=$(id -u -n)
     # the last '=' after 'user' eliminates the column header
-    local target_user=$(ps -p "${TARGET_PID}" -o user=)
+    local target_user=$(ps -o pid,user= | grep " ${TARGET_PID}$" | xargs | cut -f 2 -d " ")
```

Comment:
Hi,

This still does not work for me.

```
+ sanity_check
++ ps -o pid=
++ grep ' 93203$'
+ local pid=
+ '[' -z ']'
+ exit_on_err 1 'The target pid (93203) does not exist!'
+ [[ ! -z The target pid (93203) does not exist! ]]
+ echo 'The target pid (93203) does not exist!'
The target pid (93203) does not exist!
```

I think the leading space should be removed.

Reply:
`ps -o pid,user= | grep '93203' | cut -f 2 -d ' '` This works under macOS.

`ps -o pid,user= | grep '93203$' | cut -f 2 -d ' '` This does work under maxOS.

I think `ps -o pid,user= | grep "${TARGET_PID}" | xargs | cut -f 2 -d " "` works for me.

Reply:
How about `grep -E "^\s*${TARGET_PID}$"`? The problem with the simple `"${TARGET_PID}"` is that it matches as long as `${TARGET_PID}` is a substring of the pid.

Reply:
@Vlatombe Sorry for the late reply. `grep -E "^\s*${TARGET_PID}$"` looks good to me.